### PR TITLE
Feature validate external links

### LINKED ISSUE #164: Validate external links in documentation
We should have a test that ensures that external links in all files in `docs` (and maybe the two readmes) actually work.

### PR DESCRIPTION
**Pull request by timtroendle**
Fixes #164  .

## Checklist

Any checks which are not relevant to the PR can be pre-checked by the PR creator. All others should be checked by the reviewer. You can add extra checklist items here if required by the PR.

- [x] CHANGELOG updated
- [x] Minimal workflow tests pass
- [x] Tests added to cover contribution
- [x] Documentation updated
- [x] Configuration schema updated

### CODE REVIEW

**Comment by timtroendle** [comment]
This caught 7 broken links. Seems very valuable.

File: .github/workflows/externallinks.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Check Markdown links
+
+on:
+    push:
+    pull_request:
+    schedule:
+        - cron: "0 3 * * MON" # Runs Mondays at 3am.
```
Comment (brynpickering) [inline .github/workflows/externallinks.yaml:7]: How would we be actually notified of link issues when running weekly?

**Review by brynpickering — APPROVED** [review]
Looks good!

File: .github/workflows/externallinks.yaml (hunk 1 above)
Comment (timtroendle) [reply .github/workflows/externallinks.yaml:7]: By email. I guess it's send to all admins or whatever the right role is. For my fork, I received the email this afternoon.

**Pull request merged**